packages/language-server/src/server.ts

```diff
@@ -46,11 +53,20 @@ export function startLanguageServer(connection: vscode.Connection, env: RuntimeE
 	let hasConfigurationCapability = false;
 
 	connection.onInitialize((params: vscode.InitializeParams) => {
+		let isPnpInit = false;
 		const environment: 'node' | 'browser' = params.initializationOptions?.environment ?? 'node';
 		const workspaceUris = params.workspaceFolders?.map((folder) => folder.uri.toString()) ?? [params.rootUri ?? ''];
 
 		workspaceUris.forEach((uri) => {
 			uri = urlToPath(uri) as string;
+			if (!isPnpInit) {
+				const possiblePnpPath = getWorkspacePnpPath(uri);
+
+				if (possiblePnpPath) {
+					require(possiblePnpPath).setup();
```

Comment:
This is a security issue, the server will execute any file named `.pnp.cjs` and its `setup` method. 

I am unfamiliar with PnP, but this should be either behind an init flag or only done after a question has been asked to the user.

Reply:
That's an interesting question. My mind on the flag side, because we're talking about lsp server and I don't know how to actually show prompt to user from server's side

Reply:
I am okay with an init flag, it's *technically* possible to show a prompt and answer from the server's side (at least, VS Code can do it), but it's super cumbersome so

Reply:
Actually, I've figured out that this feature isn't so hard to implement 🙈
<img width="708" alt="image" src="https://user-images.githubusercontent.com/2384454/234670974-7464f73b-efa8-41e9-8ab2-608eca940130.png">
Thanks for feedback, I'll try this approach.

Reply:
@Princesseuh, so I've decided to use explicit init option.

Notification with actions returns a Promise, so in existing architecture it's hard to get the result without rewriting at least half of `server.ts` file 🙈

With this option i've got this in my lsp config:

<img width="951" alt="image" src="https://user-images.githubusercontent.com/2384454/234679205-d826ac3e-096c-42dd-b85d-47e5f5ccf010.png">